### PR TITLE
rd/iam_policy - tagging support

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #17594

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSIAMPolicy_'
--- PASS: TestAccAWSIAMPolicy_disappears (27.88s)
--- PASS: TestAccAWSIAMPolicy_description (40.09s)
--- PASS: TestAccAWSIAMPolicy_path (40.27s)
--- PASS: TestAccAWSIAMPolicy_namePrefix (40.36s)
--- PASS: TestAccAWSIAMPolicy_basic (40.69s)
--- PASS: TestAccAWSIAMPolicy_policy (68.83s)
--- PASS: TestAccAWSIAMPolicy_tags (91.92s)


--- PASS: TestAccAWSDataSourceIAMPolicy_basic (36.37s)
```

### CODE REVIEW

File: aws/resource_aws_iam_policy.go [hunk 1]
```diff
@@ -268,12 +293,12 @@ func iamPolicyPruneVersions(arn string, iamconn *iam.IAM) error {
 		}
 	}
 
-	err1 := iamPolicyDeleteVersion(arn, *oldestVersion.VersionId, iamconn)
+	err1 := iamPolicyDeleteVersion(arn, *oldestVersion.VersionId, conn)
```
Comment (ewbankkit) [inline aws/resource_aws_iam_policy.go:296]:
```suggestion
	err1 := iamPolicyDeleteVersion(arn, aws.StringValue(oldestVersion.VersionId), conn)
```

File: aws/resource_aws_iam_policy.go [hunk 2]
```diff
@@ -282,35 +307,35 @@ func iamPolicyDeleteNondefaultVersions(arn string, iamconn *iam.IAM) error {
 		if *version.IsDefaultVersion {
 			continue
 		}
-		if err := iamPolicyDeleteVersion(arn, *version.VersionId, iamconn); err != nil {
+		if err := iamPolicyDeleteVersion(arn, *version.VersionId, conn); err != nil {
```
Comment (ewbankkit) [inline aws/resource_aws_iam_policy.go:310]:
```suggestion
		if err := iamPolicyDeleteVersion(arn, aws.StringValue(version.VersionId), conn); err != nil {
```

File: aws/data_source_aws_iam_policy.go [hunk 3]
```diff
@@ -30,6 +30,11 @@ func dataSourceAwsIAMPolicy() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+			"policy_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"tags": tagsSchema(),
```
Comment (ewbankkit) [inline aws/data_source_aws_iam_policy.go:37]:
```suggestion
			"tags": tagsSchemaComputed(),
```

**Comment by ewbankkit** [comment]
Verified acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSIAMPolicy_\|TestAccAWSDataSourceIAMPolicy_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSIAMPolicy_\|TestAccAWSDataSourceIAMPolicy_ -timeout 180m
=== RUN   TestAccAWSDataSourceIAMPolicy_basic
=== PAUSE TestAccAWSDataSourceIAMPolicy_basic
=== RUN   TestAccAWSIAMPolicy_basic
=== PAUSE TestAccAWSIAMPolicy_basic
=== RUN   TestAccAWSIAMPolicy_description
=== PAUSE TestAccAWSIAMPolicy_description
=== RUN   TestAccAWSIAMPolicy_tags
=== PAUSE TestAccAWSIAMPolicy_tags
=== RUN   TestAccAWSIAMPolicy_disappears
=== PAUSE TestAccAWSIAMPolicy_disappears
=== RUN   TestAccAWSIAMPolicy_namePrefix
=== PAUSE TestAccAWSIAMPolicy_namePrefix
=== RUN   TestAccAWSIAMPolicy_path
=== PAUSE TestAccAWSIAMPolicy_path
=== RUN   TestAccAWSIAMPolicy_policy
=== PAUSE TestAccAWSIAMPolicy_policy
=== CONT  TestAccAWSDataSourceIAMPolicy_basic
=== CONT  TestAccAWSIAMPolicy_disappears
=== CONT  TestAccAWSIAMPolicy_namePrefix
=== CONT  TestAccAWSIAMPolicy_policy
=== CONT  TestAccAWSIAMPolicy_path
=== CONT  TestAccAWSIAMPolicy_tags
=== CONT  TestAccAWSIAMPolicy_description
=== CONT  TestAccAWSIAMPolicy_basic
--- PASS: TestAccAWSIAMPolicy_disappears (23.59s)
--- PASS: TestAccAWSDataSourceIAMPolicy_basic (26.33s)
--- PASS: TestAccAWSIAMPolicy_basic (29.82s)
--- PASS: TestAccAWSIAMPolicy_namePrefix (29.98s)
--- PASS: TestAccAWSIAMPolicy_description (29.98s)
--- PASS: TestAccAWSIAMPolicy_path (30.00s)
--- PASS: TestAccAWSIAMPolicy_policy (38.79s)
--- PASS: TestAccAWSIAMPolicy_tags (46.39s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	46.523s
```

File: .changelog/18276.txt [hunk 4]
```diff
@@ -0,0 +1,11 @@
+```release-note:enhancement
+resource/iam_policy: Add tagging support
```
Comment (bflad) [inline .changelog/18276.txt:2]:
```suggestion
resource/aws_iam_policy: Add tagging support
```

File: .changelog/18276.txt [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+```
+
+```release-note:enhancement
+resource/iam_policy: Add `policy_id` attribute
```
Comment (bflad) [inline .changelog/18276.txt:6]:
```suggestion
resource/aws_iam_policy: Add `policy_id` attribute
```

File: .changelog/18276.txt [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+```
+
+```release-note:enhancement
+data-source/iam_policy: Add `policy_id` and `tags` attributes
```
Comment (bflad) [inline .changelog/18276.txt:10]:
```suggestion
data-source/aws_iam_policy: Add `policy_id` and `tags` attributes
```

File: website/docs/r/iam_policy.html.markdown [hunk 7]
```diff
@@ -45,17 +45,19 @@ The following arguments are supported:
 * `path` - (Optional, default "/") Path in which to create the policy.
   See [IAM Identifiers](https://docs.aws.amazon.com/IAM/latest/UserGuide/Using_Identifiers.html) for more information.
 * `policy` - (Required) The policy document. This is a JSON formatted string. For more information about building AWS IAM policy documents with Terraform, see the [AWS IAM Policy Document Guide](https://learn.hashicorp.com/terraform/aws/iam-policy)
+* `tags` - Key-value mapping of tags for the IAM Policy
```
Comment (bflad) [inline website/docs/r/iam_policy.html.markdown:48]:
```suggestion
* `tags` - (Optional) Map of resource tags for the IAM Policy
```

File: aws/resource_aws_iam_policy_test.go [hunk 8]
```diff
@@ -164,6 +166,50 @@ func TestAccAWSIAMPolicy_description(t *testing.T) {
...
 
+func TestAccAWSIAMPolicy_tags(t *testing.T) {
+	var out iam.GetPolicyOutput
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_iam_policy.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSIAMPolicyDestroy,
```
Comment (bflad) [inline aws/resource_aws_iam_policy_test.go:177]:
```suggestion
		PreCheck:     func() { testAccPreCheck(t) },
		ErrorCheck:   testAccErrorCheck(t, iam.EndpointsID),
		Providers:    testAccProviders,
		CheckDestroy: testAccCheckAWSIAMPolicyDestroy,
```

**Review by bflad — APPROVED** [review]
Looks good, thank you, @DrFaust92 🚀 

Output from acceptance testing in AWS Commercial:

```
--- PASS: TestAccAWSDataSourceIAMPolicy_basic (20.45s)

--- PASS: TestAccAWSIAMPolicy_basic (22.11s)
--- PASS: TestAccAWSIAMPolicy_description (21.92s)
--- PASS: TestAccAWSIAMPolicy_disappears (18.51s)
--- PASS: TestAccAWSIAMPolicy_namePrefix (22.36s)
--- PASS: TestAccAWSIAMPolicy_path (22.27s)
--- PASS: TestAccAWSIAMPolicy_policy (29.66s)
--- PASS: TestAccAWSIAMPolicy_tags (36.52s)
```

Output from acceptance testing in AWS GovCloud (US):

```
--- PASS: TestAccAWSDataSourceIAMPolicy_basic (25.35s)

--- PASS: TestAccAWSIAMPolicy_basic (27.37s)
--- PASS: TestAccAWSIAMPolicy_description (27.37s)
--- PASS: TestAccAWSIAMPolicy_disappears (21.46s)
--- PASS: TestAccAWSIAMPolicy_namePrefix (27.86s)
--- PASS: TestAccAWSIAMPolicy_path (27.19s)
--- PASS: TestAccAWSIAMPolicy_policy (41.21s)
--- PASS: TestAccAWSIAMPolicy_tags (53.53s)
```

**Pull request merged**